src/main/java/com/google/sps/servlets/EditEventServlet.java

```diff
@@ -0,0 +1,242 @@
+
+    return event;
+  }
+
+  /**
+   * Supplies the request with attributes from the event requested.
+   *
+   * @return the request with attributes set.
+   */
+  private HttpServletRequest populateRequest(HttpServletRequest request, Entity event) {
+    String name = event.getProperty("eventName").toString();
+    String description = event.getProperty("eventDescription").toString();
+    String address = event.getProperty("address").toString();
+    String start = event.getProperty("unformattedStart").toString();
+    String end = event.getProperty("unformattedEnd").toString();
+    String date = event.getProperty("unformattedDate").toString();
+    String tags = Utils.convertToJson(event.getProperty("tags"));
+    String key = event.getProperty("eventKey").toString();
+    long eventId = event.getKey().getId();
+
+    request.setAttribute("name", name);
+    request.setAttribute("description", description);
+    request.setAttribute("date", date);
+    request.setAttribute("start", start);
+    request.setAttribute("end", end);
+    request.setAttribute("address", address);
+    request.setAttribute("tags", tags);
+    request.setAttribute("key", key);
+    request.setAttribute("id", eventId);
+    request.setAttribute("tags", tags);
+    request.setAttribute("key", key);
+    request.setAttribute("id", eventId);
+
+    return request;
+  }
+
+  /**
+   * Retrieves the key string (if any) from the request parameter and converts it to type Key.
+   *
+   * @return the key from the request parameter.
+   */
+  private Key getEventKey(HttpServletRequest request, String name)
+      throws IllegalArgumentException, IOException {
+    Key eventKey = null;
+
+    // Get the string from the request.
+    if (request.getParameter(name) != null) {
+      String eventKeyString = request.getParameter(name);
+
+      // Convert String to type Key.
+      eventKey = KeyFactory.stringToKey(eventKeyString);
+    } else {
+      throw new IOException("Request is missing parameter");
+    }
+    return eventKey;
+  }
+
+  /**
+   * Checks if the event queried belongs to the user logged in.
+   *
+   * @return a boolean 'exists' which contains if the event belongs to the user
+   */
+  private boolean isUsersEvent(String userID, Key entityKey) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String eventKeyString = KeyFactory.keyToString(entityKey);
+    boolean exists = false;
+
+    // Query events created by user logged in.
+    Query query =
+        new Query("Event")
+            .setFilter(new Query.FilterPredicate("creator", Query.FilterOperator.EQUAL, userID));
+    PreparedQuery queried = datastore.prepare(query);
+
+    // Checks if the event is in the list of events created by the user.
+    for (Entity e : queried.asIterable()) {
+      String userEventKey = e.getProperty("eventKey").toString();
+
+      if (userEventKey.equals(eventKeyString)) {
+        exists = true;
+      }
+    }
```

Comment:
Would it be more efficient if we queried for the event we're trying to edit using its key and then checked if it's creator is the current user?

Reply:
I agree!

Reply:
Yes it definitely would! Thanks for pointing that out.